File: Altair/scatterplot.py
import altair as alt
import pandas as pd

# Load the penguins dataset
url = 'https://raw.githubusercontent.com/Rob-hub-lang/a2-DataVis-5Ways/main/penglings.csv'
df = pd.read_csv(url)

# Define custom colors for each species
species_colors = {'Adelie': 'darkorange', 'Chinstrap': 'darkorchid', 'Gentoo': 'darkgreen'}

# Create scatter plot
scatter_plot = alt.Chart(df).mark_circle(opacity=0.8).encode(
    x=alt.X('flipper_length_mm:Q', scale=alt.Scale(zero=False)),
    y=alt.Y('body_mass_g:Q', scale=alt.Scale(zero=False)),
    color=alt.Color('species:N', scale=alt.Scale(domain=list(species_colors.keys()), range=list(species_colors.values()))),
    size=alt.Size('bill_length_mm:Q'),
    tooltip=['species:N', 'flipper_length_mm:Q', 'body_mass_g:Q', 'bill_length_mm:Q']
).properties(
    width=800,
    height=500
)

# Add axes with ticks and labels
x_axis = alt.Axis(title='Flipper Length (mm)')
y_axis = alt.Axis(title='Body Mass (g)')
scatter_plot = scatter_plot.configure_axis(
    labelFontSize=12,
    titleFontSize=14,
    tickCount=10
).encode(
    x=alt.X('flipper_length_mm:Q', scale=alt.Scale(zero=False), axis=x_axis),
    y=alt.Y('body_mass_g:Q', scale=alt.Scale(zero=False), axis=y_axis)
)

# Display the plot
scatter_plot.save('scatter_plot.html')

File: MatLab/scatterplot.m
% Load the penguins dataset
url = 'https://raw.githubusercontent.com/Rob-hub-lang/a2-DataVis-5Ways/main/penglings.csv';
data = readtable(url);

% Extract relevant columns
flipperLength = data.flipper_length_mm;
disp(sum(isnan(flipperLength)));
bodyMass = data.body_mass_g;
disp(sum(isnan(bodyMass)));
billLength = data.bill_length_mm;
species = categorical(data.species);

% Set up figure
figure;

% Loop through each species and create a scatter plot
hold on; % Allow multiple plots on the same axes
speciesList = categories(species);
colors = [1 0.647 0; 0.545 0.373 0.694; 0 0.545 0.545]; % RGB values for 'darkorange', 'darkorchid', 'darkgreen'
for i = 1:length(speciesList)
    currentSpecies = speciesList{i};
    indices = species == currentSpecies;
    scatter(flipperLength(indices), bodyMass(indices), billLength(indices)*5, 'MarkerFaceColor', colors(i, :), ...
        'DisplayName', currentSpecies, 'MarkerEdgeAlpha', 0.8, 'MarkerFaceAlpha', 0.8, 'SizeData', billLength(indices)*5);
end
hold off;

% Customize axes
xlabel('Flipper Length (mm)');
ylabel('Body Mass (g)');
title('Scatter Plot of Penguins');
grid on;

% Add legend with explicit entries
legend('Location', 'best');

% Add colorbar for bill length
c = colorbar;
c.Label.String = 'Bill Length (mm)';
caxis([min(billLength), max(billLength)]);

% Adjust figure properties
set(gca, 'FontSize', 12);
set(gcf, 'Position', [100, 100, 800, 600]);

% Save the plot as an image (optional)
saveas(gcf, 'scatter_plot_matlab.png');
